packages/atlas/src/components/_ypp/DetailsFormDialog/DetailsFormDialog.tsx

```diff
@@ -0,0 +1,125 @@
+
+import { FormFieldsWrapper } from './DetailsFormDialog.styles'
+
+export const DetailsFormDialog: FC = () => {
+  const [foundChannel, setFoundChannel] = useState<BasicChannelFieldsFragment | null>()
+  const [loading, setLoading] = useState(false)
+  const client = useApolloClient()
+  const {
+    setValue,
+    register,
+    control,
+    formState: { errors },
+  } = useForm<{ email: string | undefined; channel: string | undefined }>({ mode: 'onChange' })
+
+  const debounceFetchChannels = useRef(
+    debouncePromise(async (value?: string) => {
+      if (!value) {
+        return
+      }
+      try {
+        const {
+          data: { channels },
+        } = await client.query<GetBasicChannelsQuery, GetBasicChannelsQueryVariables>({
+          query: GetMembershipsDocument,
+          variables: { where: { title_eq: value } },
+        })
+        setFoundChannel(channels.length ? channels[0] : undefined)
+      } catch (error) {
+        SentryLogger.error('Failed to fetch memberships', 'WhiteListTextField', error)
+      } finally {
+        setLoading(false)
+      }
+    }, 500)
+  )
+  return (
+    <FormFieldsWrapper>
+      <FormField
+        label="Contact email"
+        description="We need your email address to send you payment information. No spam or marketing materials."
+        error={errors.email?.message}
+      >
+        <Controller
+          control={control}
+          name="email"
+          rules={{
+            validate: {
+              required: (value) => {
+                if (!value || !value.length) {
+                  return 'Enter contact email.'
+                }
+              },
+              valid: (value) => {
+                if (value && !EMAIL_PATTERN.test(value)) {
+                  return 'Enter valid email address.'
+                }
+                return true
+              },
+            },
+          }}
+          render={({ field: { value, onChange } }) => (
+            <Input placeholder="Email address" type="email" value={value} onChange={onChange} error={!!errors.email} />
+          )}
+        />
+      </FormField>
+      <FormField
+        optional
+        label="Referrer"
+        description="Enter the handle of the Joystream channel which recommended the program to you."
+        error={errors.channel?.message}
+      >
+        <Input
+          {...register('channel', {
+            validate: {
+              required: (value) => {
+                if (value && value.length && !foundChannel && !loading) {
+                  return 'No member with this handle has been found.'
+                }
+                return true
+              },
+            },
+            onChange: (event) => {
+              const { value } = event.target
+              setLoading(true)
+              setValue('channel', event.target.value, { shouldTouch: true, shouldDirty: true })
+              if (value.length) {
+                debounceFetchChannels.current(value)
```

Comment:
How this validation actually works? I think errors just randomly show up, and doesn't matter if the channel exists or not

https://user-images.githubusercontent.com/51168865/190166749-81e6b132-cd06-45a1-ba31-830a95c6a417.mov

<shameless_plug>I think you could look at the code in the create membership form. The debounced validation works pretty well there </shameless_plug>